arctic/store/version_store.py

```diff
@@ -710,21 +714,28 @@ def restore_version(self, symbol, as_of, prune_previous_version=True):
         `VersionedItem`
             VersionedItem named tuple containing the metadata of the written symbol's version document in the store.
         """
-        # if version/snapshot/data supplied in "as_of" does not exist, will fail fast with NoDataFoundException
+        # TODO: This operation could be optimized further by not doing a read-write, but instead do meta only update
+        #       and only do concat_rewrite on the first append operation, after the restore (i.e. using a flag).
+        #       Keep in mind the history branch induced corruption (see Issue 
+
+        # Don't create a new version if we are trying to restore to the latest
+        # TODO: this is not atomic, and is prone to race conditions if multiple processes touch the same symbol
         version_to_restore = self._read_metadata(symbol, as_of=as_of)
+        try:
+            latest_version = self._read_metadata(symbol)
+            if version_to_restore['version'] == latest_version['version']:
+                return latest_version
```

Comment:
Is this actually doing a restore - it looks like it's just returning the version?

Reply:
this is a noop really, i.e. when we try to restore to a snapshot/version which is the latest (no write/append/delete/write_metadata since then), then do nothing, and don't cause an unnecessary read/write